bam-cmd/src/main/java/org/victorchang/QnameCommand.java

```diff
@@ -149,15 +149,15 @@ private boolean createDirectory(Path path, boolean force) {
     @Option(names = {"-f", "--file-name"}, description = "Path to a file containing QNAMEs to search for (separated by newline).")
     Path qnamePath;
 
-    List<String> getQnames() throws IllegalArgumentException {
+    Set<String> getQnames() throws IllegalArgumentException {
         if (qname != null) {
-            return singletonList(qname);
+            return Set.of(qname);
         }
-        final Map<Boolean, List<String>> qnames = readFile()
+        final Map<Boolean, Set<String>> qnames = readFile()
                     .stream()
                     .filter(StringUtils::isNotBlank)
-                    .collect(partitioningBy(this::isValidQname));
-            final List<String> invalidQnames = qnames.getOrDefault(false, emptyList());
+                    .collect(partitioningBy(this::isValidQname, toSet()));
```

Comment:
We should collect to `LinkedHashSet` to preserve the order of QNames of the input file - this will make the tool a bit more predictable

Reply:
It is a little more tricky than that. Since we sort the qnames for efficient lookup we probably need to maintain the initial order separately before printing out to achieve this. It can be done as a separate change.